web/federate.go

```diff
@@ -0,0 +1,98 @@
+package web
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"bitbucket.org/ww/goautoneg"
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/text"
+
+	"github.com/prometheus/prometheus/promql"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type Federation struct {
+	QueryEngine *promql.Engine
+}
+
+func (fed *Federation) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	qs := fmt.Sprintf("{%s}", strings.Join(req.Form["match[]"], ","))
```

Comment:
I want to be able to do something like
{job="prometheus",__metric__="up"} or {__metric__=~"job:.*"}
as there could be a few different sets of data I want that can't be specified in one selector.